astropy/units/format/base.py

```diff
@@ -61,7 +62,21 @@ def format_exponential_notation(
         str
             The value in exponential notation in a this class's format.
         """
-        return format(val, format_spec)
+        x = format(val, format_spec).split("e")
+        if len(x) != 2:
+            return cls._format_mantissa(x[0])  # no exponent
+        ex = x[1].lstrip("0+")
+        if not ex:
+            return cls._format_mantissa(x[0])  # exponent was zero
+        if ex.startswith("-"):
+            ex = "-" + ex[1:].lstrip("0")
+        ex = f"10{cls._format_superscript(ex)}"
+        m = cls._format_mantissa("" if x[0].rsplit("0") == "1." else x[0])
```

Comment:
It is a bit strange that the line here takes care of replacing `1.`, `1.0` `1.00` etc. with an empty string instead of letting `_format_mantissa()` do that, but we have quite a few `_format_mantissa()` methods and they all assume that something else performs this step.

Reply:
Shouldn't this have been `x[0].rstrip("0")`?

Reply:
Of course this should use `rstrip()` and not `rsplit()`. The fact that the tests didn't catch this means that the condition for the `if`-statement is always `False` in our tests, but it's not obvious to me if that's because the ternary is unnecessary here or if our test coverage is not high enough.